hms-app/src/common/types.tsx

```diff
@@ -1,20 +1,50 @@
+export type HackathonPreview = {
+  id: string
+  title: string
+}
+
+export type IdeaPreviews = {
+  id: string
+  title: string
+}
+
 export type Idea = {
   id: string
+  owner: Owner[]
+  hackathon: HackathonPreview[]
+  participants: Participant[]
   title: string
   description: string
-  author?: User
-  reason: string
   problem: string
   goal: string
-  skills: string[]
-  category: string
-  participants: User[]
-  aws: boolean
+  requiredSkills: Skill[]
+  category: Category[]
+  creationDate: string
 }
 
-export type User = {
+export type Category = {
+  id: string
+  title: string
+}
+
+export type Skill = {
```

Comment:
you might want to specify this (and many of the other things in here) as `XxxPreview` so that it is obvious that they don't contain all of the possible data